### PR TITLE
New article: Introduction to Audio Processing in NLP

### PR DESCRIPTION
**Pull request by spectre900**
An introduction to the various techniques used for processing audio in NLP

### CODE REVIEW

File: _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Any machine learning task involves three steps, data collection, training and evaluation. However for training a machine learning model, we cannot use raw data, we need to pre-process the data to some suitable form and extract features which can be used for training the model. In case of NLP audio signals are used as data, hence one must be familiar with the various processing techniques specific to the audio data types. One of the state of the art tool for extracting features from audio signal is MFCCs (Mel-frequency cepstral coefficients), hence we will be specifically focusing on MFCC feature extraction.
+
+
+## Useful Techniques and Terminologies
+
+
+#### Fourier Transform
+
+Fourier transform is an important tool used in signal processing, it is basically is a mathematical transform that converts a time domain signal into a frequency domain signal. When we calculate a Fourier transform, we begin with a function of time, f(t), and through mathematical decomposition, we produce a function of frequency, F(ω). When discrete signals are involved, Discrete Fourier Transform (DFT) is used which is normally computed using the so-called Fast Fourier Transform (FFT).
```
Comment (geekswaroop) [inline _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md]: Typo in first line

Comment (geekswaroop) [reply _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md]: Consider using latex for representing functions

Comment (geekswaroop) [reply _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md]: Link a resource for knowing more about FFT.

File: _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+#### Spectrum and Cepstrum
+
+Two important features in audio processing are Spectrum and Cepstrum. Both Spectrum and Cepstrum are closely related to each other.
+
+1. A spectrum is the Fourier transform of a signal, hence a spectrum is basically the frequency domain representation of a time domain audio signal.
+
+2. A cepstrum is defined as the Fourier transform of the logarithm of the spectrum. This results in a signal that's neither in the frequency domain nor in the time domain. The domain of the resulting signal is called the quefrency.
```
Comment (geekswaroop) [inline _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md:47]: Appropriate wiki links for quefrency

File: _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+The reason for converting signals into their frequency domain is related closely to the biology of the human ear. The cochlea is a portion of the inner ear that looks like a snail shell and is a fluid filled part with thousands of tiny hairs that are connected to nerves. The shorter hairs resonate with higher frequencies and the longer hairs resonate with lower frequencies. Since our ears are basically frequency analyzers, decomposing audio signals into frequency domain seems like a logical approach to extract features from it.
+
+
+#### The Mel-Frequency Scale
+
+The human hearing is hgihly selective to lower frequencies ( < 1000 Hz) and this keeps decreasing as the frequency increases. Mel-Frequency Scale is a kind of pyscho-acoustic scale, derived from a set of experiments on human subjects. It has high resolution at lower frequencies and the resolution keeps decreasing as the frequency increases, hence it helps to simulate the way human ears work.
```
Comment (geekswaroop) [inline _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md]: Typo

**Review by geekswaroop — CHANGES_REQUESTED** [review]
Please do the necessary changes suggested. Thanks!

File: _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+librosa.display.specshow(mfccs, x_axis='time')          # plotting the MFCCs
+plt.colorbar()
+plt.title('MFCC')
+plt.tight_layout()
+plt.show()
+```
+
+![MFCC Plot of Sample Audio Clip](/blog/assets/img/Audio-Processing-NLP/mfcc.png)
+
+## References
```
Comment (geekswaroop) [inline _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md:92]: - Considering adding some images representing audio signals

**Review by geekswaroop — CHANGES_REQUESTED** [review]

File: _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md (hunk 1 above)
Comment (spectre900) [reply _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md]: Done
used &omega; and added link for FFT.

File: _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md (hunk 2 above)
Comment (spectre900) [reply _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md]: Done.
Added link for cepstrum and quefrency.

File: _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md (hunk 4 above)
Comment (spectre900) [reply _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md:103]: Done.
Added wave plot of sample audio clip.

File: _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md (hunk 3 above)
Comment (spectre900) [reply _posts/2021-01-4-Introduction-to-audio-processing-in-NLP.md]: Corrected.

**Review by geekswaroop — APPROVED** [review]
LGTM, Good Job!

**Pull request merged**